pkg/source/reconciler/mtsource/kafkasource.go

```diff
@@ -120,14 +124,14 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1beta1.KafkaSource
 	defer c.Close()
 	src.Status.MarkConnectionEstablished()
 
-	err = client.InitOffsets(ctx, c, src.Spec.Topics, src.Spec.ConsumerGroup)
+	totalPartitions, err := client.InitOffsets(ctx, c, src.Spec.Topics, src.Spec.ConsumerGroup)
 	if err != nil {
 		logging.FromContext(ctx).Errorw("unable to initialize consumergroup offsets", zap.Error(err))
 		src.Status.MarkInitialOffsetNotCommitted("OffsetsNotCommitted", "Unable to initialize consumergroup offsets: %v", err)
 		return err
 	}
 	src.Status.MarkInitialOffsetCommitted()
-
+	src.Status.MaxAllowedVReplicas = totalPartitions*r.MaxEventPerSecondPerPartition/r.VReplicaMPS + 1
```

Comment:
by default there should be no limits. I would make MaxEventPerSecondPerPartition and VReplicaMPS optional and default to -1. If set then set MaxAllowedVReplicas, otherwise do nothing (no max)